benchmark/benchmark.py

```diff
@@ -55,68 +56,79 @@ def plot(title, kernels, labels, n_vals):
     import perfplot
 
     perfplot.show(
-        setup=lambda n: n,
-        kernels=kernels,
-        labels=labels,
-        n_range=n_vals,
-        xlabel="n",
-        title=title,
-        # logx="auto",  # set to True or False to force scaling
-        # logy="auto",
-        # equality_check=numpy.allclose,  # set to None to disable "correctness" assertion
-        # automatic_order=True,
-        # colors=None,
-        # target_time_per_measurement=1.0,
-        # time_unit="s",  # set to one of ("auto", "s", "ms", "us", or "ns") to force plot units
-        # relative_to=1,  # plot the timings relative to one of the measurements
-        # flops=lambda n: 3*n,  # FLOPS plots
+        setup=lambda n: n, kernels=kernels, labels=labels, n_range=n_vals, xlabel="n", title=title,
     )
 
 
-def profile(func, identifier, *, repeat=20):
+def profile(func, identifier, *, min_time=5):
     """Profile the given function.
 
     Args:
         func (Callable[[], Any]): function to profile
         identifier (str): identifying part of the name of the file containing the results
-        repeat (int): the number of times the function is run
+        min_time (float): func is called repeatedly until at least this many seconds have elapsed
     """
     import cProfile
     import pstats
 
+    print("Minimum duration: {} seconds.".format(min_time))
     pr = cProfile.Profile()
     pr.enable()
 
-    for _ in range(repeat):
+    t0 = time.process_time()
+    repeats = 0
+    while True:
         func()
+        repeats += 1
+        elapsed = time.process_time() - t0
+        if elapsed > min_time:
```

Comment:
```suggestion
        if time.process_time() - t0 > min_time:
```
Not sure how much savings you get by skipping an assignment, but I think it's important to reduce overhead in the benchmark suite itself whenever possible

Reply:
Might even be slightly smarter to put this directly as the `while` loop conditional